framework/src/preference/private_preference.c

```diff
@@ -198,6 +217,29 @@ int preference_get_string(const char *key, char **value)
 	return ret;
 }
 
+int preference_get_binary(const char *key, void **value, int *len)
+{
+	int ret;
+	preference_data_t data;
+
+	if (key == NULL) {
+		return PREFERENCE_INVALID_PARAMETER;
+	}
+
+	data.key = key;
+	data.type = PRIVATE_PREFERENCE;
+	data.attr.type = PREFERENCE_TYPE_BINARY;
+
+	/* Get preference with prctl */
+	ret = prctl(PR_GET_PREFERENCE, &data);
+	if (ret == OK) {
+		*value = (void *)data.value;
+		*len = (int *)data.attr.len;
```

Comment:
How about len for null?

Reply:
I feel that NULL values need not be checked as per my reply to the previous comment.
Kindly let me know.